File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/determine-if-a-cell-is-reachable-at-a-given-time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int lSteps = Math.max(Math.abs(fx-sx), Math.abs(fy-sy));
        int mSteps = Math.abs(fx-sx) + Math.abs(fy-sy);
        if(t < lSteps) return false;
        if(t - lSteps != 1) return true;
        return Math.abs(fx-sx) >= 1 || Math.abs(fy-sy) >= 1;
    }
}

File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/determine-if-a-cell-is-reachable-at-a-given-time.js
/**
 * @param {number} sx
 * @param {number} sy
 * @param {number} fx
 * @param {number} fy
 * @param {number} t
 * @return {boolean}
 */
var isReachableAtTime = function(sx, sy, fx, fy, t) {
    if(sx === fx && sy === fy && t === 1) return false;
    return Math.max(Math.abs(fx-sx), Math.abs(fy-sy)) <= t;
};